src/main/java/com/iheartradio/m3u8/ParseUtil.java

```diff
@@ -54,17 +54,20 @@ public static float parseFloat(String string, String tag) throws ParseException
 
         if (matcher.matches()) {
             String valueString = matcher.group(1);
+            if (valueString.length() % 2 != 0) {
+                throw ParseException.create(ParseExceptionType.INVALID_HEXADECIMAL_STRING, tag, hexString);
+            }
 
-            for (char c : valueString.toCharArray()) {
-                bytes.add(hexCharToByte(c));
+            for (int i = 0; i < valueString.length(); i += 2) {
+                bytes.add((byte)(Short.parseShort(valueString.substring(i, i+2), 16) & 0xFF));
```

Comment:
The bug description seems to be affecting the writing, not the parsing. Is there a problem with `hexCharToByte`?

Reply:
I see, the issue is 2 hex chars fit within 1 byte. I don't work with low level very often. :( I think I'm okay with this implementation.

Reply:
I noticed issues with parsing after I created the issue and updated the PR, I can open up another issue if you want

Reply:
No need, this is fine.